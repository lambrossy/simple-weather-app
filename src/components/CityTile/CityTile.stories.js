import React from "react";
import CityTile from "./CityTile";

export default {
  title: "Example/CityTile",
  component: CityTile,
};

const Template = (args) => <CityTile {...args} />;

export const Basic = Template.bind({});
Basic.args = {
  id: 2163355,
  name: "Hobart",
  description: "few clouds",
  temperature: { current: 11.28, max: 12.37, min: 9.51 },
  icon: {
    url: "http://openweathermap.org/img/wn/02n@2x.png",
    description: "Clouds",
  },
};
