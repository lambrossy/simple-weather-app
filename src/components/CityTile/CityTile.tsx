import { memo, FC } from "react";
import GlassPane from "../GlassPane/GlassPane";
import styled from "styled-components";
import round from "lodash/fp/round";
import upperFirst from "lodash/fp/upperFirst";

import CityWeather from "../../types/CityWeather";

const Container = styled.div({
  width: 240,
  height: 240,
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between",
  padding: 15,
  fontFamily: '"Georama", sans-serif',
});

const CityName = styled.h2({
  margin: 0,
  fontSize: 30,
  overflow: "hidden",
  textOverflow: "ellipsis",
  whiteSpace: "nowrap",
});

const Temperature = styled.div({
  fontSize: 60,
});

const Conditions = styled.div({
  display: "flex",
  justifyContent: "space-between",
  alignItems: "center",
});

const Image = styled.img({
  margin: "-20px -10px",
});

const Description = styled.h2({
  margin: 0,
  fontSize: 24,
  overflow: "hidden",
  textOverflow: "ellipsis",
  whiteSpace: "nowrap",
});

const TemperatureRange = styled.div({
  display: "flex",
  justifyContent: "space-between",
  fontSize: 24,
});

const CityTile: FC<CityWeather> = ({
  name,
  icon,
  temperature,
  description,
}) => (
  <GlassPane>
    <Container>
      <CityName>{name}</CityName>
      <Conditions>
        <Temperature>{round(temperature.current)}°</Temperature>
        <Image alt={icon.description} src={icon.url} />
      </Conditions>
      <Description>{upperFirst(description)}</Description>
      <TemperatureRange>
        <div>Low: {round(temperature.min)}°</div>
        <div>High: {round(temperature.max)}°</div>
      </TemperatureRange>
    </Container>
  </GlassPane>
);

export default memo(CityTile);
