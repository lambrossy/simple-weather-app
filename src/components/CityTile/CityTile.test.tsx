import { render } from "@testing-library/react";
import CityTile from "./CityTile";

const props = {
  id: "2163355",
  name: "Hobart",
  description: "few clouds",
  temperature: { current: 11.28, max: 12.37, min: 9.51 },
  icon: {
    url: "http://openweathermap.org/img/wn/02n@2x.png",
    description: "Clouds",
  },
};

it("renders correctly", () => {
  const element = render(<CityTile {...props} />);
  expect(element.container).toMatchSnapshot();
});
