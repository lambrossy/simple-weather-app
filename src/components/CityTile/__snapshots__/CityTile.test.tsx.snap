// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-bdnxRM jcNukz"
  >
    <div
      class="sc-gtsrHT iMbPwy"
    >
      <h2
        class="sc-dlnjwi efKzcR"
      >
        Hobart
      </h2>
      <div
        class="sc-eCApnc dTKwPB"
      >
        <div
          class="sc-hKFxyN fyrwTC"
        >
          11
          °
        </div>
        <img
          alt="Clouds"
          class="sc-jSFjdj hNba-Dv"
          src="http://openweathermap.org/img/wn/02n@2x.png"
        />
      </div>
      <h2
        class="sc-gKAaRy eQsPIB"
      >
        Few clouds
      </h2>
      <div
        class="sc-iCoGMd eXYBxk"
      >
        <div>
          Low: 
          10
          °
        </div>
        <div>
          High: 
          12
          °
        </div>
      </div>
    </div>
  </div>
</div>
`;
